pandas/core/series.py

```diff
@@ -1786,8 +1786,18 @@ def sort_index(self, axis=0, level=None, ascending=True, inplace=False,
             indexer = _ensure_platform_int(indexer)
             new_index = index.take(indexer)
         else:
-            new_index, indexer = index.sort_values(return_indexer=True,
-                                                   ascending=ascending)
+            from pandas.core.groupby import _nargsort
+
+            if ((ascending and index.is_monotonic_increasing) or
```

Comment:
I would do this optimization inside ``_nargsort``. you can try and see if anything breaks the test suite.

this would just return an indexer like ``np.arange(len(index))``.